internal/controller/controllers/crd_utils_test.go

```diff
@@ -0,0 +1,65 @@
+package controllers
+
+import (
+	"context"
+
+	"github.com/golang/mock/gomock"
+	"github.com/google/uuid"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/internal/controller/api/v1alpha1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	fakeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+var _ = Describe("create agent CR", func() {
+	var (
+		c                client.Client
+		ctx              = context.Background()
+		mockCtrl         *gomock.Controller
+		clusterName      = "test-cluster"
+		clusterNamespace = "test-namespace"
+		crdUtils         *CRDUtils
+		log              = common.GetTestLog().WithField("pkg", "controllers")
+	)
+
+	BeforeEach(func() {
+		c = fakeclient.NewFakeClientWithScheme(scheme.Scheme)
+		crdUtils = NewCRDUtils(c)
+		mockCtrl = gomock.NewController(GinkgoT())
+	})
+
+	AfterEach(func() {
+		mockCtrl.Finish()
+	})
+
+	Context("create agent", func() {
+
+		It("create agent success", func() {
+			id := uuid.New().String()
+			agent := newHost(id, clusterNamespace, v1alpha1.AgentSpec{})
+			err := crdUtils.CreateAgentCR(ctx, log, id, clusterNamespace, clusterName)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(c.Create(ctx, agent)).Should(HaveOccurred())
```

Comment:
Maybe i missed something, but shouldn't `c.Create(ctx, agent)` be called from `CreateAgentCR` 
And the test need to do a get and see that the resource actually exists?

Reply:
Creating the same CR twice will throw an error that it already exists.
I will do a Read instead to make it more clear.

Reply:
So yes, i missed `Should(HaveOccurred())`